From 06a24c4e6ec4d876bb5c9216a6b5f5364d58dc10 Mon Sep 17 00:00:00 2001
From: Jiffin Tony Thottan <jthottan@redhat.com>
Date: Fri, 28 Apr 2017 17:27:46 +0530
Subject: [PATCH 58/74] glusterd/ganesha : perform removal of ganesha.conf on
 nodes only in ganesha cluster

Change-Id: I864ecd9391adf80fb1fa6ad2f9891a9ce77135e7
Signed-off-by: Jiffin Tony Thottan <jthottan@redhat.com>
Reviewed-on: https://review.gluster.org/17138
Smoke: Gluster Build System <jenkins@build.gluster.org>
Reviewed-by: soumya k <skoduri@redhat.com>
NetBSD-regression: NetBSD Build System <jenkins@build.gluster.org>
CentOS-regression: Gluster Build System <jenkins@build.gluster.org>
Reviewed-by: Kaleb KEITHLEY <kkeithle@redhat.com>
---
 xlators/mgmt/glusterd/src/glusterd-ganesha.c | 21 +++++++++++----------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/xlators/mgmt/glusterd/src/glusterd-ganesha.c b/xlators/mgmt/glusterd/src/glusterd-ganesha.c
index 38fa378..2392341 100644
--- a/xlators/mgmt/glusterd/src/glusterd-ganesha.c
+++ b/xlators/mgmt/glusterd/src/glusterd-ganesha.c
@@ -740,17 +740,18 @@ stop_ganesha (char **op_errstr) {
         int ret                 = 0;
         runner_t runner         = {0,};
 
-        runinit (&runner);
-        runner_add_args (&runner,
-                         GANESHA_PREFIX"/ganesha-ha.sh",
-                         "--setup-ganesha-conf-files", CONFDIR, "no", NULL);
-        ret =  runner_run (&runner);
-        if (ret) {
-                gf_asprintf (op_errstr, "removal of symlink ganesha.conf "
-                             "in /etc/ganesha failed");
-        }
-
         if (check_host_list ()) {
+                runinit (&runner);
+                runner_add_args (&runner,
+                                 GANESHA_PREFIX"/ganesha-ha.sh",
+                                 "--setup-ganesha-conf-files", CONFDIR,
+                                 "no", NULL);
+                ret =  runner_run (&runner);
+                if (ret) {
+                        gf_asprintf (op_errstr, "removal of symlink ganesha.conf "
+                                     "in /etc/ganesha failed");
+                }
+
                 ret = manage_service ("stop");
                 if (ret)
                         gf_asprintf (op_errstr, "NFS-Ganesha service could not"
-- 
1.8.3.1

